src/service_daemon.rs

```diff
@@ -1199,21 +1211,19 @@ impl Zeroconf {
         self.my_services.insert(service_fullname, info);
     }
 
-    /// Sends out annoucement of `info` on every valid interface.
-    /// Returns the list of interface IPs that sent out the annoucement.
+    /// Sends out announcement of `info` on every valid interface.
+    /// Returns the list of interface IPs that sent out the announcement.
     fn send_unsolicited_response(&self, info: &ServiceInfo) -> Vec<IpAddr> {
         let mut outgoing_addrs = Vec::new();
-        let mut subnet_set: HashSet<u128> = HashSet::new();
+        // Send the announcement on one interface per ip version.
+        let mut intf_idx_ip_ver_set: HashSet<(u32, u8)> = HashSet::new();
 
         for (_, intf_sock) in self.intf_socks.iter() {
-            let subnet = ifaddr_subnet(&intf_sock.intf.addr);
-            if !intf_sock.intf.is_link_local() && subnet_set.contains(&subnet) {
-                continue; // no need to send again in the same subnet.
+            if intf_idx_ip_ver_set.contains(&intf_sock.intf_idx_ip_ver()) {
```

Comment:
Related to a previous comment, I think we probably should check something like:
`if intf_sock.intf_idx_ip_ver().0 != 0`